test/functional/ui/output_spec.lua

```diff
@@ -22,6 +22,7 @@ describe("shell command :!", function()
 
   after_each(function()
     screen:detach()
+    session.command('call jobstop(b:terminal_job_id)')
```

Comment:
sending ctrl-c in `after_each` seems to fix the problem:

    child_session.feed_data("\3")  -- ctrl-c